src/plugins/scopes/ScopesApi.js

```diff
@@ -0,0 +1,474 @@
+/* @flow */
+import type Element from '../../elements/Element';
+import Node from '../../elements/Node';
+import Program from '../../elements/types/Program';
+import BlockStatement from '../../elements/types/BlockStatement';
+import Identifier from '../../elements/types/Identifier';
+import Scope from './Scope';
+import AssignmentPattern from '../../elements/types/AssignmentPattern';
+import FunctionExpression from '../../elements/types/FunctionExpression';
+import FunctionDeclaration from '../../elements/types/FunctionDeclaration';
+import ArrowFunctionExpression from '../../elements/types/ArrowFunctionExpression';
+import VariableDeclarator from '../../elements/types/VariableDeclarator';
+import VariableDeclaration from '../../elements/types/VariableDeclaration';
+import AssignmentExpression from '../../elements/types/AssignmentExpression';
+import UpdateExpression from '../../elements/types/UpdateExpression';
+import ForOfStatement from '../../elements/types/ForOfStatement';
+import ForInStatement from '../../elements/types/ForInStatement';
+import MemberExpression from '../../elements/types/MemberExpression';
+import Property from '../../elements/types/Property';
+import MethodDefinition from '../../elements/types/MethodDefinition';
+import ImportDefaultSpecifier from '../../elements/types/ImportDefaultSpecifier';
+import ImportNamespaceSpecifier from '../../elements/types/ImportNamespaceSpecifier';
+import ImportSpecifier from '../../elements/types/ImportSpecifier';
+import ThisExpression from '../../elements/types/ThisExpression';
+import Super from '../../elements/types/Super';
+import CatchClause from '../../elements/types/CatchClause';
+import LabeledStatement from '../../elements/types/LabeledStatement';
+import BreakStatement from '../../elements/types/BreakStatement';
+import ContinueStatement from '../../elements/types/ContinueStatement';
+import ClassExpression from '../../elements/types/ClassExpression';
+import ClassDeclaration from '../../elements/types/ClassDeclaration';
+import JSXIdentifier from '../../elements/types/JSXIdentifier';
+import JSXAttribute from '../../elements/types/JSXAttribute';
+import JSXElement from '../../elements/types/JSXElement';
+import JSXMemberExpression from '../../elements/types/JSXMemberExpression';
+import JSXNamespacedName from '../../elements/types/JSXNamespacedName';
+import {types} from './Definition';
+
+const scopedBlocks = {
+    'ForStatement': true,
+    'ForInStatement': true,
+    'ForOfStatement': true,
+    'SwitchStatement': true,
+    'CatchClause': true
+};
+
+export default class ScopesApi {
+    constructor(program: Program) {
+        program.on('elements-add', (elements: Element[]) => {
+            elements.map(this._addElement, this);
+        });
+        program.on('elements-remove', (elements: Element[]) => {
```

Comment:
I mean you could omit parentheses. Or go even further, like 
```js
program.on('elements-remove', (elements: Element[]) => elements.map(() => this._removeElement));
```

Although i wouldn't :)